spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/store/JdbcTokenStore.java

```diff
@@ -129,12 +129,14 @@ public OAuth2AccessToken mapRow(ResultSet rs, int rowNum) throws SQLException {
 			LOG.error("Could not extract access token for authentication " + authentication, e);
 		}
 
-		if (accessToken != null
-				&& !key.equals(authenticationKeyGenerator.extractKey(readAuthentication(accessToken.getValue())))) {
-			removeAccessToken(accessToken.getValue());
-			// Keep the store consistent (maybe the same user is represented by this authentication but the details have
-			// changed)
-			storeAccessToken(accessToken, authentication);
+		if (accessToken != null) {
+			OAuth2Authentication oldAuthentication = readAuthentication(accessToken.getValue());
+			if (oldAuthentication == null || !key.equals(authenticationKeyGenerator.extractKey(oldAuthentication))) {
```

Comment:
@MikeN123 I'm feeling like we should not apply this change. It hides the fact that the persistent store contains invalid / incompatible data. Given that the Spring Security libraries have been upgraded, it is expected that these deserialization errors would occur if the stored version is incompatible and therefore should be revealed to the application via an error instead of hiding it altogether.

I believe some form of cleanup or migration would need to happen to the persistent store if Spring Security libraries are upgraded.

Reply:
I disagree. There are all kinds of places where the project already does automatic cleanup. It even does that in this place - `readAuthentication()` will silently remove the stored data, but log a warning, and the next request will succeed. The fact that a NPE is thrown is just an error, in other places it will happily continue (e.g., earlier in this method it retrieves `accessToken`, if `deserializeAccessToken()` fails there, it returns `null` as well, which is subsequently checked and not throwing some kind of error.

It is undoable as a developer to clean up the persistent store at the right times, as it is almost impossible to keep track of when Spring Security does or does not make a change that changes the serialization version of their classes (especially when you pull in a Spring Security upgrade through a Spring Boot upgrade).

If it should be doing something different here, then at least it should just return `null` on deserialization failure, like it does when the result of `selectAccessTokenFromAuthenticationSql` fails to deserialize. Otherwise it's just inconsistent with all other places where deserialization failures silently remove the stored data.

There is also not really something that the application can do about such failures, except for truncating the store. This is used in the `token` endpoint which is directly exposed to the outside world, it is not like the library user could be doing some error handling themselves.

Reply:
@MikeN123 Apologies as my initial assessment of the changes was incorrect. This looks good and I'll have it merged.

Reply:
Np, thanks :)